### PR TITLE
Refactor Bitcoin transactions API

### LINKED ISSUE #255: Support parsing txs contain Bitcoin coinbase input
zcoin sample json
```
{
	"txid": "792a3568a084c59ef42462aa2a3aa44c5f8709c75b305a0884d7e6c585086c8e",
	"version": 1,
	"vin": [{
		"n": 0,
		"isAddress": false,
		"hex": "",
		"coinbase": "0000000000000000000000000000000000000000000000000000000000000000"
	}],
	"vout": [{
		"value": "100000000",
		"n": 0,
		"spent": true,
		"hex": "76a914526ac6dd84927e5e26dc6c89976d58ea4c1a4d4788ac",
		"addresses": ["a8EF4cpenEgEn9hm2NL5KfFK1UmSZZaQVn"],
		"isAddress": true
	}],
	"blockHash": "9fac8ebaace7a81a7c74192d97b92fd081c583149b36d8d5463694b0ac18a87d",
	"blockHeight": 128068,
	"confirmations": 53260,
	"blockTime": 1547640057,
	"value": "100000000",
	"valueIn": "0",
	"fees": "0",
	"hex": ""
}

```

### LINKED ISSUE #256: Return transaction direction for Bitcoin txs
Currently, we decide `outgoing`, `incoming`, `self` in client code (iOS/Android/Desktop), it would be nice to return the direction from API, especially for UTXO coins

### LINKED ISSUE #257: Sum all possible outputs when calculating transfer value
Client logic: https://github.com/trustwallet/ios/blob/master/Trust/Blockchains/Clients/Bitcoin/Models/BitcoinTransactionDetail.swift#L93

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #255
Fixes #256 
Fixes #257

### CODE REVIEW

File: platform/bitcoin/api.go [hunk 1]
```diff
@@ -5,8 +5,8 @@ import (
 	"github.com/spf13/viper"
 	"github.com/trustwallet/blockatlas"
 	"github.com/trustwallet/blockatlas/coin"
+	"github.com/deckarep/golang-set"
```
Comment (madcake) [inline platform/bitcoin/api.go]: How is it library?

File: platform/bitcoin/api.go [hunk 2]
```diff
@@ -64,68 +64,103 @@ func (p *Platform) getTxsByXPub(xpub string) ([]blockatlas.Tx, error) {
...
-	for _, transfer := range transfers {
-		for _, address := range transfer.Addresses {
-			if strings.EqualFold(address, originAddress) {
-				return true
+func parseOutputs(outputs []Output) (addresses []string) {
+	set := make(map[string]bool)
+	result := []string{}
+	for _, output := range outputs {
+		for _, address := range output.Addresses {
+			if set[address] {
```
Comment (madcake) [inline platform/bitcoin/api.go]: Maybe more easy
```golang
if !set[address] {

}
```

Comment (madcake) [reply platform/bitcoin/api.go]: Why you check that? Set should consists on from unique values.

File: platform/bitcoin/client.go [hunk 3]
```diff
@@ -14,36 +17,45 @@ type Client struct {
...
 	url := fmt.Sprintf("%s/v2/tx/%s", c.URL, id)
 	resp, err := c.HTTPClient.Get(url)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 
-	var receipt TransferReceipt
-	err = json.NewDecoder(resp.Body).Decode(&receipt)
+	var transaction Transaction
```
Comment (vikmeup) [inline platform/bitcoin/client.go]: take a look at waves clients, how to send and handle requests.

File: platform/bitcoin/api.go (hunk 1 above)
Comment (hewigovens) [reply platform/bitcoin/api.go]: A simple set type for the Go language. Also used by Docker, 1Password, Ethereum.

File: platform/bitcoin/client.go (hunk 3 above)
Comment (hewigovens) [reply platform/bitcoin/client.go]: We can always improve, here I just renamed variables. BTW, you should do a careful review earlier, the namings were nonsense, get txs by address didn't work too

File: platform/bitcoin/api.go (hunk 2 above)
Comment (hewigovens) [reply platform/bitcoin/api.go]: To keep the orders

**Comment by hewigovens** [comment]
@madcake @vikmeup take a look again

**Pull request merged**